Create SPI for Enhanced Security

darryl-apple opened the pull request:
#### 5a9f1e86f8d88f1187ec2a58b6ce805036091ad1
<pre>
Create WKAPI for EnhancedSecurity
<a href="https://bugs.webkit.org/show_bug.cgi?id=298942">https://bugs.webkit.org/show_bug.cgi?id=298942</a>
<a href="https://rdar.apple.com/160683091">rdar://160683091</a>

Reviewed by NOBODY (OOPS!).

Explanation of why this fixes the bug (OOPS!).

Tests: security/cocoa/check-enhanced-security-disabled.html
       security/cocoa/check-enhanced-security-enabled.html
       security/cocoa/enhanced-security/check-enhanced-security-enabled.html
       Tools/TestWebKitAPI/SourcesCocoa.txt
       Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm
* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/security/cocoa/check-enhanced-security-disabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-disabled.html: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled.html: Added.
* Source/WebCore/page/Page.h:
(WebCore::Page::setEnhancedSecurityEnabled):
(WebCore::Page::enhancedSecurityEnabled const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::enhancedSecurityEnabled const):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
(WebKit::XPCServiceInitializer):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::XPCServiceInitializerDelegate::getExtraInitializationData):
(WebKit::setJSCOptions):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::enhancedSecurityEnabled const):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isEnhancedSecurityEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _setEnhancedSecurityEnabled:]):
(-[WKWebpagePreferences _enhancedSecurityEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
(WebKit::BrowsingContextGroup::sharedProcessForSite):
* Source/WebKit/UIProcess/BrowsingContextGroup.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::isEnhancedSecurityEnabled):
(WebKit::WebPageProxy::shouldEnableEnhancedSecurity const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForSite):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::prepareProcessForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_toolbarsAreVisible):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
(WebKit::WebProcess::isEnhancedSecurityEnabled):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Tools/MiniBrowser/mac/SettingsController.h:
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController _populateMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController toggleEnhancedSecurityEnabled:]):
(-[SettingsController enhancedSecurityEnabled]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController didChangeSettings]):
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldEnableEnhancedSecurity):
(WTR::hardcodedFeaturesBasedOnPathForTest):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm: Added.
(TestWebKitAPI::isEnhancedSecurityEnabled):
(TestWebKitAPI::isJITEnabled):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnablesTrue)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnableFalse)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityDisablesJIT)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityNavigationStaysEnabledAfterNavigation)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecurity)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySamePage)):
(TestWebKitAPI::psonProcessPoolConfiguration):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPool)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPoolReverse)):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enhancedSecurityEnabled const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):
</pre>

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dd79e365999450663564962c219cf8a5744c3991. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/584ca719e959123efef54ee8e5793ebb2c83bed0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/eb31c56e4b9b69b498db0e3253641453f0349d8c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

weinig commented:
While I don't want to comment on the naming of API or SPI exposed by Apple for WebKit.framework, I would love for a more descriptive name than "enhanced security" to be used internally in the engine. "enhanced security" seems quite vague.

Ideally in the engine, this would be broken down into specific settings that can have clear names.

pvollan commented on Source/WebCore/page/Page.h:
```diff
@@ -1845,6 +1848,8 @@ class Page : public RefCountedAndCanMakeWeakPtr<Page>, public Supplementable<Pag
     bool m_modelLoadDelaysDisabledForTesting { false };
 #endif
+
+    bool m_isEnhancedSecurityEnabled { false };
```
Nit: `m_isEnhancedSecurityEnabled ` -> `m_enhancedSecurityEnabled `

pvollan commented on Source/WebCore/loader/FrameLoaderTypes.h:
```diff
@@ -172,6 +172,11 @@ enum class AllowsContentJavaScript : bool {
     Yes,
 };
 
+enum class EnhancedSecurity : bool {
```
Nit: `EnhancedSecurity` -> `EnhancedSecurityEnabled`

darryl-apple replied:
Done

pvollan commented on Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
```diff
@@ -194,13 +198,14 @@ void setOSTransaction(OSObjectPtr<os_transaction_t>&& transaction)
     if (xpc_dictionary_get_bool(initializerMessage, "configure-jsc-for-testing"))
         JSC::Config::configureForTesting();
     if (enableLockdownMode == EnableLockdownMode::Yes) {
+        RELEASE_ASSERT(enableEnhancedSecurity == EnableEnhancedSecurity::No);
```
Should we enable Enhanced Security in Lockdown Mode? Perhaps this will enabled in a coming PR?

darryl-apple replied:
Please see the changes in XPCServiceEntryPoint.mm. I've made it so only LockdownMode or EnhancedSecurity can be enabled with LockdownMode taking precedence.

pvollan commented on Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
```diff
@@ -367,6 +367,13 @@ bool PageConfiguration::lockdownModeEnabled() const
     return lockdownModeEnabledBySystem();
 }
 
+bool PageConfiguration::enhancedSecurity() const
```
Nit: `enhancedSecurity` -> `enhancedSecurityEnabled`

darryl-apple replied:
Done

pvollan commented on Source/WebKit/UIProcess/API/APIPageConfiguration.h:
```diff
@@ -441,6 +441,8 @@ class PageConfiguration : public ObjectImpl<Object::Type::PageConfiguration> {
     bool isLockdownModeExplicitlySet() const;
     bool lockdownModeEnabled() const;
     
+    bool enhancedSecurity() const;
```
Ditto.

pvollan commented on Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
```diff
@@ -112,6 +112,9 @@ class WebsitePolicies final : public API::ObjectImpl<API::Object::Type::WebsiteP
     WebCore::AllowsContentJavaScript allowsContentJavaScript() const { return m_data.allowsContentJavaScript; }
     void setAllowsContentJavaScript(WebCore::AllowsContentJavaScript allows) { m_data.allowsContentJavaScript = allows; }
 
+    WebCore::EnhancedSecurity enhancedSecurity() const { return m_enhancedSecurityEnabled; }
+    void setEnhancedSecurity(WebCore::EnhancedSecurity enabled) { m_enhancedSecurityEnabled = enabled; }
```
Nit: `enhancedSecurity` -> `enhancedSecurityEnabled`

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e27fa867aec3d48c178f87d37992ea702b2328cf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

pvollan commented on Source/WebCore/page/Page.h:
```diff
@@ -1376,6 +1376,9 @@ class Page : public RefCountedAndCanMakeWeakPtr<Page>, public Supplementable<Pag
 #endif
 
+    void setEnhancedSecurityEnabled(bool enabled) { m_enhancedSecurityEnabled = enabled; };
+    bool enhancedSecurityEnabled() { return m_enhancedSecurityEnabled; };
```
const

pvollan commented on Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
```diff
@@ -212,6 +218,10 @@ void setJSCOptions(xpc_object_t initializerMessage, EnableLockdownMode enableLoc
             JSC::Options::allowDoubleShape() = false;
             JSC::Options::alwaysHaveABadTime() = true;
         });
+    } else if (enableEnhancedSecurity == EnableEnhancedSecurity::Yes) {
+        JSC::Options::initialize([] {
+            JSC::ExecutableAllocator::disableJIT();
+        });
     } else if (xpc_dictionary_get_bool(initializerMessage, "disable-jit")) {
         JSC::Options::initialize([] {
             JSC::ExecutableAllocator::disableJIT();
```
We could perhaps share the code here.

darryl-apple commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -6560,6 +6560,17 @@ bool Internals::audioSessionActive() const
     return false;
 }
 
+bool Internals::isEnhancedSecurityEnabled() const
```
Done

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2692579d845c0c8871dc3b0047bc4dcadbc44d4b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

pvollan commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
```diff
@@ -323,6 +323,7 @@ for this property.
 - (void)_executeEditCommand:(NSString *)command argument:(NSString *)argument completion:(void (^)(BOOL))completion WK_API_AVAILABLE(macos(10.13.4), ios(11.3));
 
 - (void)_isJITEnabled:(void(^)(BOOL))completionHandler WK_API_AVAILABLE(macos(10.14.4), ios(12.2));
+- (void)_isEnhancedSecurityEnabled:(void(^)(BOOL))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
```
Should this be API, or is it better to keep as SPI?

darryl-apple replied:
It's only used in tests so my understanding is that SPI should be fine

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b0a2c8029977d46a9adc206a53986a43b12c5b26. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

darryl-apple commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -6238,6 +6238,7 @@ enum class WebCore::ColorSchemePreference : uint8_t {
 enum class WebCore::ModalContainerObservationPolicy : bool
 
 enum class WebCore::AllowsContentJavaScript : bool
+enum class WebCore::EnhancedSecurityEnabled : bool
```
Good spot. We don't need this enum anymore so I have removed. I have then refactored to use a shared enum in the WebKit namespace

pvollan commented on Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
```diff
@@ -112,6 +113,9 @@ class WebsitePolicies final : public API::ObjectImpl<API::Object::Type::WebsiteP
     WebCore::AllowsContentJavaScript allowsContentJavaScript() const { return m_data.allowsContentJavaScript; }
     void setAllowsContentJavaScript(WebCore::AllowsContentJavaScript allows) { m_data.allowsContentJavaScript = allows; }
 
+    WebKit::EnhancedSecurityEnabled enhancedSecurityEnabled() const { return m_enhancedSecurityEnabled; }
+    void setEnhancedSecurityEnabled(WebKit::EnhancedSecurityEnabled enabled) { m_enhancedSecurityEnabled = enabled; }
```
I wonder if we should just use `bool` here. Then we'd avoid having two similar enums for Enhanced Security. What do you think?

pvollan commented on Source/WebKit/Shared/EnhancedSecurityEnabled.h:
```diff
@@ -0,0 +1,32 @@
+
+namespace WebKit {
+
+enum class EnhancedSecurityEnabled : bool { Disabled, Enabled };
```
It looks like it is being used in WebProcessPool.

I'd suggest renaming to

`enum class EnhancedSecurity: bool { Disabled, Enabled }`

darryl-apple replied:
yes it's still used in the following:
`bool shouldEnableEnhancedSecurity() const { return m_enhancedSecurity == EnhancedSecurityEnabled::Enabled; }`
I've kept the enum as this mirrors the code LockdownMode:

darryl-apple replied:
good spot, yes it should be moved back into WebProcessProxy, now we have removed the other usages.
Done

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/847c860e2cb425bcc83c113dee812c5932a028eb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/740f86d41be865d86262a0b1812995c1eacadfde. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4abf722d2a2deedaa8a8e1429603a6d5ff5b5e75. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

pvollan commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -705,6 +705,7 @@ void WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess(Remo
 
     RefPtr requestingProcess = requestingProcessIdentifier ? WebProcessProxy::processForIdentifier(*requestingProcessIdentifier) : nullptr;
     auto lockdownMode = requestingProcess ? requestingProcess->lockdownMode() : (lockdownModeEnabledBySystem() ? WebProcessProxy::LockdownMode::Enabled : WebProcessProxy::LockdownMode::Disabled);
+    auto enhancedSecurityEnabled = requestingProcess ? requestingProcess->enhancedSecurity() : WebProcessProxy::EnhancedSecurity::Disabled;
```
Nit: `enhancedSecurityEnabled` -> `enhancedSecurity`

pvollan commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -1110,7 +1114,8 @@ void WebProcessPool::prewarmProcess()
 
     auto lockdownMode = lockdownModeEnabledBySystem() ? WebProcessProxy::LockdownMode::Enabled : WebProcessProxy::LockdownMode::Disabled;
-    createNewWebProcess(nullptr, lockdownMode, WebProcessProxy::IsPrewarmed::Yes);
+    auto enhancedSecurityEnabled = WebProcessProxy::EnhancedSecurity::Disabled;
```
Ditto.

pvollan commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -1292,6 +1297,7 @@ Ref<WebPageProxy> WebProcessPool::createWebPage(PageClient& pageClient, Ref<API:
 
     RefPtr<WebProcessProxy> process;
     auto lockdownMode = pageConfiguration->lockdownModeEnabled() ? WebProcessProxy::LockdownMode::Enabled : WebProcessProxy::LockdownMode::Disabled;
+    auto enhancedSecurityEnabled = pageConfiguration->enhancedSecurityEnabled() ? WebProcessProxy::EnhancedSecurity::Enabled : WebProcessProxy::EnhancedSecurity::Disabled;
```
Ditto.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a936dd4ae2472692b1d560e064d1a5e78855bf28. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50798

darryl-apple committed:
Create SPI for Enhanced Security
https://bugs.webkit.org/show_bug.cgi?id=298942
rdar://160683091

Reviewed by Per Arne Vollan.

Create new SPI for Enhanced Security

Tests: security/cocoa/check-enhanced-security-disabled.html
       security/cocoa/check-enhanced-security-enabled.html
       security/cocoa/enhanced-security/check-enhanced-security-enabled.html
       Tools/TestWebKitAPI/SourcesCocoa.txt
       Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm
       Tools/TestWebKitAPI/Tests/WebKitCocoa/TextExtractionTests.mm
* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/security/cocoa/check-enhanced-security-disabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-disabled.html: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/check-enhanced-security-enabled.html: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled-expected.txt: Added.
* LayoutTests/security/cocoa/enhanced-security/check-enhanced-security-enabled.html: Added.
* Source/WebCore/page/Page.h:
(WebCore::Page::setEnhancedSecurityEnabled):
(WebCore::Page::enhancedSecurityEnabled const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::enhancedSecurityEnabled const):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
(WebKit::XPCServiceInitializer):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::XPCServiceInitializerDelegate::getExtraInitializationData):
(WebKit::setJSCOptions):
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::enhancedSecurityEnabled const):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _isJITEnabled:]):
(-[WKWebView _isEnhancedSecurityEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _setEnhancedSecurityEnabled:]):
(-[WKWebpagePreferences _enhancedSecurityEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
(WebKit::BrowsingContextGroup::sharedProcessForSite):
* Source/WebKit/UIProcess/BrowsingContextGroup.h:
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationActionPolicyDecision):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::isEnhancedSecurityEnabled):
(WebKit::WebPageProxy::shouldEnableEnhancedSecurity const):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::establishRemoteWorkerContextConnectionToNetworkProcess):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForSite):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::prepareProcessForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
(WebKit::WebProcessProxy::getLaunchOptions):
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_toolbarsAreVisible):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeProcess):
(WebKit::WebProcess::isEnhancedSecurityEnabled):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/WebProcess.messages.in:
* Tools/MiniBrowser/mac/SettingsController.h:
* Tools/MiniBrowser/mac/SettingsController.m:
(-[SettingsController _populateMenu:]):
(-[SettingsController validateMenuItem:]):
(-[SettingsController toggleEnhancedSecurityEnabled:]):
(-[SettingsController enhancedSecurityEnabled]):
* Tools/MiniBrowser/mac/WK2BrowserWindowController.m:
(-[WK2BrowserWindowController didChangeSettings]):
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldEnableEnhancedSecurity):
(WTR::hardcodedFeaturesBasedOnPathForTest):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/EnhancedSecurity.mm: Added.
(TestWebKitAPI::isEnhancedSecurityEnabled):
(TestWebKitAPI::isJITEnabled):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnablesTrue)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityEnableFalse)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityDisablesJIT)):
(TestWebKitAPI::TEST(EnhancedSecurity, EnhancedSecurityNavigationStaysEnabledAfterNavigation)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecurity)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySamePage)):
(TestWebKitAPI::psonProcessPoolConfiguration):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPool)):
(TestWebKitAPI::TEST(EnhancedSecurity, PSONToEnhancedSecuritySharedProcessPoolReverse)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/TextExtractionTests.mm:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enhancedSecurityEnabled const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):

Canonical link: https://commits.webkit.org/300264@main

webkit-commit-queue commented:
Committed 300264@main (093c69553b5f): <https://commits.webkit.org/300264@main>

Reviewed commits have been landed. Closing PR #50798 and removing active labels.
